src/main/java/uk/ac/cam/cl/dtg/segue/search/IsaacSearchInstructionBuilder.java

```diff
@@ -0,0 +1,332 @@
+            if (contentType.equals(EVENT_TYPE) && !includePastEvents) {
+                LocalDate today = LocalDate.now();
+                long now = today.atStartOfDay(ZoneId.systemDefault()).toEpochSecond() * Constants.EVENT_DATE_EPOCH_MULTIPLIER;
+                contentInstruction.must(new RangeMatchInstruction<Long>(Constants.DATE_FIELDNAME).greaterThanOrEqual(now));
+            }
+
+            // Apply instructions to search for specific terms in specific fields
+            this.addSearchesInFieldsToInstruction(contentInstruction, searchesInFields);
+
+            contentInstruction.setMinimumShouldMatch(minimumMatchInstructions);
+
+            if (this.priorityIncludedContentTypes.contains(contentType)) {
+                contentInstruction.setBoost(PRIORITY_CONTENT_BOOST);
+            }
+
+            this.masterInstruction.should(contentInstruction);
+        }
+        return this.masterInstruction;
+    }
+
+    /**
+     * Augments {@code instructions} with the field search instructions specified via {@code searchInField()} and friends.
+     *
+     * @param instructions The master BooleanMatchInstruction to augment with the field-search instructions.
+     * @param searchesInFields A list of {@code SearchInField}s encapsulating fields and terms to search for, as well as
+     *                         the strategy and priority to use for each.
+     */
+    private void addSearchesInFieldsToInstruction(final BooleanMatchInstruction instructions, final List<SearchInField> searchesInFields) {
+
+        Map<String, NestedMatchInstruction> nestedQueriesByPath = new HashMap<>();
+
+        for (SearchInField searchInField : searchesInFields) {
+
+            // Special fields
+            if (Arrays.stream(Constants.ADDRESS_PATH_FIELDNAME).collect(Collectors.toList()).contains(searchInField.getField())) {
+                // Address fields
+                // If we're searching any "address" fields, search all of them
+                String nestedFieldConnector = searchProvider.getNestedFieldConnector();
+                String addressPath = String.join(nestedFieldConnector, Constants.ADDRESS_PATH_FIELDNAME);
+
+                for (String addressField : Constants.ADDRESS_FIELDNAMES) {
+                    for (String term : searchInField.getTerms()) {
+                        String field = addressPath + nestedFieldConnector + addressField;
+                        instructions.should(new ShouldMatchInstruction(field, term, EVENT_ADDRESS_FIELD_BOOST,
+                                false));
+                        instructions.should(new ShouldMatchInstruction(field, term, EVENT_ADDRESS_FIELD_BOOST_FUZZY,
+                                true));
+                    }
+                }
+            } else if (Constants.NESTED_FIELDS.contains(searchInField.getField())) {
+                // Nested fields
+                // Ensure nested field queries are grouped together
```

Comment:
I'd recommend adding a link to https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-nested-query.html, or similar, as Nested queries are "special". I would also be tempted to rename Constants.NESTED_FIELDS to Constants.NESTED_QUERY_FIELDS to try to help differentiate between them and the "qualified" field query like we have for address search.